trino/test_trino.py

```diff
@@ -0,0 +1,126 @@
+            table, ",".join(["({})".format(x % 2) for x in range(400)]))
+        self.assert_instance_command(
+            name, "hive --database {} -e '{}'".format(schema, query))
+
+    def __validate_data_in_table_via_trino(self, name, table, schema):
+        query = "SELECT number, count(*) AS total FROM {} GROUP BY number ORDER BY number DESC;".format(
+            table)
+        _, stdout, _ = self.assert_instance_command(
+            name,
+            "trino --catalog=hive --schema={} --execute='{}' --output-format TSV"
+            .format(schema, query))
+        self.assertEqual(stdout, "1\t200\n0\t200\n")
+
+    def __verify_coordinators_count(self, name, coordinators, server_param=""):
+        query = "select count(*) from system.runtime.nodes where coordinator=true"
+        _, stdout, _ = self.assert_instance_command(
+            name, "trino {} --execute '{}' --output-format TSV".format(
+                server_param, query))
+        self.assertEqual(
+            coordinators, int(stdout),
+            "Bad number of coordinators. Expected: {}\tFound: {}".format(
+                coordinators, stdout))
+
+    def __verify_workers_count(self, name, workers, server_param=""):
+        query = "select count(*) from system.runtime.nodes where coordinator=false"
+        _, stdout, _ = self.assert_instance_command(
+            name, "trino {} --execute '{}' --output-format TSV".format(
+                server_param, query))
+        self.assertEqual(
+            workers, int(stdout),
+            "Bad number of workers. Expected: {}\tFound: {}".format(
+                workers, stdout))
+
+    @parameterized.parameters(
+        ("SINGLE", ["m"], 1, 0),
+        ("STANDARD", ["m"], 1, 2),
+        ("HA", ["m-0"], 1, 2),
+    )
+    def test_trino(self, configuration, machine_suffixes, coordinators,
+                    workers):
+        if self.getImageOs() == 'centos':
+            self.skipTest("Not supported in CentOS-based images")
+
+        # Skip on 2.0+ version of Dataproc because it's not supported
+        if self.getImageVersion() >= pkg_resources.parse_version("2.0"):
+            self.skipTest("Not supported in 2.0+ images")
+
+        self.createCluster(configuration, self.INIT_ACTIONS)
+        for machine_suffix in machine_suffixes:
+            self.verify_instance(
+                "{}-{}".format(self.getClusterName(), machine_suffix),
+                coordinators, workers)
+
+    @parameterized.parameters(("SINGLE", ["m"], 1, 0))
+    def test_trino_custom_port(self, configuration, machine_suffixes,
+                                coordinators, workers):
+        if self.getImageOs() == 'centos':
+            self.skipTest("Not supported in CentOS-based images")
+
+        # Skip on 2.0+ version of Dataproc because it's not supported
+        if self.getImageVersion() >= pkg_resources.parse_version("2.0"):
+            self.skipTest("Not supported in 2.0+ images")
```

Comment:
What are the limitations? Dataproc 1.X will no longer be supported by the end of 2022.

Reply:
Ah, this is an obsolete code piece, there is no limitation and works fine with 2+ images.

Reply:
Can you check?